action.go

```diff
@@ -280,6 +191,173 @@ func (c *cmdAction) String() string {
+		// we can just take half of remaining arguments as keys. If the number of IDs does not match the number of
+		// keys the command will fail later when send to Redis so no need for us to handle that case.
+		ids := len(args[i+1:]) / 2
+
+		return args[i+1 : len(args)-ids]
+	}
+
+	return nil
+}
+
+func (cfg CmdConfig) actionProperties(cmd string, args ...string) ActionProperties {
+	pf := cfg.ActionProperties
+	if pf == nil {
+		pf = defaultActionProperties
+	}
+	return pf(cmd, args...)
+}
+
+// Cmd works like the global Cmd function but using the value returned by
+// calling the ActionProperties callback specified in cfg (or a builtin default
+// if the callback is nil) with cmd and args as parameters, using the returned
+// properties for the newly created Action.
+//
+// Calling Cmd on a zero-value CmdConfig is the same as calling the global Cmd
+// function.
+//
+// See the documentation for the global Cmd function for information about the
+// returned Action.
+func (cfg CmdConfig) Cmd(rcv interface{}, cmd string, args ...string) Action {
+	c := getCmdAction()
+	*c = cmdAction{
+		rcv:  rcv,
+		cmd:  cmd,
+		args: args,
+	}
+	c.properties = cfg.actionProperties(c.cmd, c.args...)
+	return c
+}
+
+// FlatCmd works like the global FlatCmd function but using the value returned
+// by calling the ActionProperties callback specified in cfg (or a builtin
+// default if the callback is nil) with cmd and args as parameters, using the
+// returned properties for the newly created Action.
+//
+// Calling FlatCmd on a zero-value CmdConfig is the same as calling the global
+// FlatCmd function.
+//
+// See the documentation for the global FlatCmd function for information about
+// the returned Action.
+func (cfg CmdConfig) FlatCmd(rcv interface{}, cmd string, args ...interface{}) Action {
+	c := getCmdAction()
+	*c = cmdAction{
+		rcv: rcv,
+		cmd: cmd,
+	}
+	c.args, c.flattenErr = resp3.Flatten(args, nil)
+	c.properties = cfg.actionProperties(c.cmd, c.args...)
+	return c
+}
+
+// Cmd is used to perform a redis command and retrieve a result. It should not
+// be passed into Do more than once.
+//
+// If the receiver value of Cmd is nil then the result is discarded.
+//
+// If the receiver value of Cmd is a primitive, a slice/map, or a struct then a
+// pointer must be passed in. It may also be an io.Writer, an
+// encoding.Text/BinaryUnmarshaler, or a resp.Unmarshaler.
+//
+// The Action returned by Cmd also implements resp.Marshaler.
+//
+// The ActionProperties for the Action method will be automatically determined
+// based on the command used. For non-builtin commands (e.g. provided by Redis
+// Modules) the ActionProperties may not always return correct results.
+//
+// In this case it's possible to create a new CmdConfig instance with a custom
+// ActionProperties resolver and creating the Action via the CmdConfig`s Cmd
+// method.
+//
+// See the documentation on CmdConfig for more information.
```

Comment:
I think the previous two paragraphs (`The ActionProperties for the Action method...` and `In this case...`) could be omitted, and this line could just say `See CmdConfig's documentation if more configurability is required, e.g. if using commands provided by Redis Modules`.

Same with on FlatCmd.

Reply:
Done